src/status_im2/contexts/communities/overview/view.cljs

```diff
@@ -82,23 +83,23 @@
                  (+ 38 (int (Math/ceil (layout-y %))))
                  (into #{} (map (comp :name second) channels-list)))
     :style     {:margin-top 8 :flex 1}}
-   (doall
-    (for [[category-id {:keys [chats name collapsed?]}] channels-list]
-      [rn/view
-       {:key       category-id
-        ;; on-layout fires only when the component re-renders, so
-        ;; in case the category hasn't changed, it will not be fired
-        :on-layout #(on-category-layout name (int (layout-y %)))}
-       (when-not (= constants/empty-category-id category-id)
-         [quo/divider-label
-          {:on-press     #(collapse-category community-id category-id collapsed?)
-           :chevron-icon (if collapsed? :i/chevron-right :i/chevron-down)
-           :chevron      :left}
-          name])
-       (when-not collapsed?
-         (into [rn/view {:style {:padding-horizontal 8 :padding-bottom 8}}]
-               (map #(channel-chat-item community-id community-color %))
```

Comment:
FYI: `doall` is unnecessary and Reagent components shouldn't be called as functions, this can cause problems.

Reply:
@ilmotta 
> Reagent components shouldn't be called as functions, this can cause problems.

I disagree with this, it depends on what you want to achieve.
We convert a function into a Regent component when we call it using `[]` instead of `()`.
In this case, this piece of code is just mapping a function over chats to return hiccup that looks like:

```clojure
[rn/view {:style {:padding-horizontal ,,,}}
  [rn/view [quo/channel ,,,]] ; <-- begins the result of mapping
  [rn/view [quo/channel ,,,]]
  [rn/view [quo/channel ,,,]]
  ,,,
  ]
```
So I see no problem that could be caused here.

If you want to change the implementation to return a sequence with the `:key` metadata, maybe (IDK) you are getting a React warning about duplicating keys, since the hiccup returned with this change is:

```clojure
[rn/view {:style {:padding-horizontal ,,,}}
 (
  ^{:key id-1} [channel-chat-item ,,,]
  ^{:key id-2} [channel-chat-item ,,,]
  ^{:key id-3} [channel-chat-item ,,,]
  ,,,
 )
]
```
and the new Reagent component `channel-chat-item` returns:

```clojure
[rn/view {:key id} ; <-- this is using the same id
  [quo/channel ,,,]
```

so we have two components with the same id

Reply:
Correct @ulisesmac, here it's not a problem I believe, that's why I said *can cause problems*. I still prefer to almost never treat component functions as normal functions, unless maybe when playing in the REPL or doing something more advanced.

Although there's more than one way to do the same thing, I don't see a reason to call the component as a function and I prefer hiccup all the way whenever possible.